vision/failure_flags.py

```diff
@@ -0,0 +1,60 @@
+
+class FailureFlags:
+    """
+    Abstract class used by the pipeline to track which algorithm failed
+    in the event of the program crashing
+
+    For each attribute, a value of True signals that the algorithm
+    ran to completion without any problems
+    """
+    def __init__(self):
+        pass
```

Comment:
While you are already making more changes, you could also add a string function to the base class that says something like "INVALID STATE".